packages/mermaid/src/mermaidAPI.ts

```diff
@@ -374,219 +378,17 @@ export const removeExistingElements = (
-  const graphInit = utils.detectInit(text);
-  if (graphInit) {
-    directiveSanitizer(graphInit);
-    configApi.addDirective(graphInit);
-  }
-
-  const config = configApi.getConfig();
-  log.debug(config);
-
-  // Check the maximum allowed text size
-  // TODO: Remove magic number
-  if (text.length > (config?.maxTextSize ?? 50000)) {
-    text = MAX_TEXTLENGTH_EXCEEDED_MSG;
-  }
-
-  // clean up text CRLFs
-  text = text.replace(/\r\n?/g, '\n'); // parser problems on CRLF ignore all CR and leave LF;;
-
-  const idSelector = '#' + id;
-  const iFrameID = 'i' + id;
-  const iFrameID_selector = '#' + iFrameID;
-  const enclosingDivID = 'd' + id;
-  const enclosingDivID_selector = '#' + enclosingDivID;
-
-  let root: any = select('body');
-
-  const isSandboxed = config.securityLevel === SECURITY_LVL_SANDBOX;
-  const isLooseSecurityLevel = config.securityLevel === SECURITY_LVL_LOOSE;
 
-  const fontFamily = config.fontFamily;
-
-  // -------------------------------------------------------------------------------
-  // Define the root d3 node
-  // In regular execution the svgContainingElement will be the element with a mermaid class
-
-  if (svgContainingElement !== undefined) {
-    if (svgContainingElement) {
-      svgContainingElement.innerHTML = '';
-    }
-
-    if (isSandboxed) {
-      // If we are in sandboxed mode, we do everything mermaid related in a (sandboxed )iFrame
-      const iframe = sandboxedIframe(select(svgContainingElement), iFrameID);
-      root = select(iframe.nodes()[0]!.contentDocument!.body);
-      root.node().style.margin = 0;
-    } else {
-      root = select(svgContainingElement);
-    }
-    appendDivSvgG(root, id, enclosingDivID, `font-family: ${fontFamily}`, XMLNS_XLINK_STD);
-  } else {
-    // No svgContainingElement was provided
-
-    // If there is an existing element with the id, we remove it. This likely a previously rendered diagram
-    removeExistingElements(document, id, enclosingDivID, iFrameID);
-
-    // Add the temporary div used for rendering with the enclosingDivID.
-    // This temporary div will contain a svg with the id == id
-
-    if (isSandboxed) {
-      // If we are in sandboxed mode, we do everything mermaid related in a (sandboxed) iFrame
-      const iframe = sandboxedIframe(select('body'), iFrameID);
-      root = select(iframe.nodes()[0]!.contentDocument!.body);
-      root.node().style.margin = 0;
-    } else {
-      root = select('body');
-    }
-
-    appendDivSvgG(root, id, enclosingDivID);
-  }
-
-  text = encodeEntities(text);
-
-  // -------------------------------------------------------------------------------
-  // Create the diagram
-
-  // Important that we do not create the diagram until after the directives have been included
-  let diag;
-  let parseEncounteredException;
-
-  try {
-    // diag = new Diagram(text);
-    diag = getDiagramFromText(text);
-    if ('then' in diag) {
```

Comment:
I noticed that the if('then' in diag) { ... on line 476 on current file is no longer present. Not needed any longer?

Reply:
Nope, that was used to check whether it's a promise or not. Now, it'll always be a promise. 
Nice catch!